web/src/app/browser/src/languageMenu.ts

```diff
@@ -0,0 +1,582 @@
+      const obs = this.originalBodyStyle;
+      const dbs = document.body.style;
+      obs.overflowY = dbs.overflowY;
+      obs.height = dbs.height;
+
+      // Now that the properties we're going to overwrite have been cached...
+      dbs.overflowY = 'hidden';
+      dbs.height = '100%';
+      return true;
+    }
+
+    const unlockBodyScroll = () => {
+      if(!this.originalBodyStyle) {
+        // We shouldn't be able to reach here, but in case things go out-of-order due
+        // to some unforeseen circumstance, let's null-guard here.
+        console.error("Unexpected state:  `originalBodyStyle` is unset; cannot restore original body style");
+        return;
+      }
+
+      // Reverses the changes to document.body.style made by `lockBodyScroll`.
+      const obs = this.originalBodyStyle;
+      const dbs = document.body.style;
+
+      dbs.overflowY = obs.overflowY;
+      dbs.height = obs.height;
+
+      // Successful restoration!  Clear the "restore to this" state so that the
+      // next 'lock' operation knows to do its part.
+      this.originalBodyStyle = null;
+    }
+
+    // Touchstart (or mspointerdown) event highlights the touched list item
+    const touchStart=function(e) {
+      e.stopPropagation();
+      if(this.className.indexOf('selected') <= 0) {
+        this.className=this.className+' selected';
+      }
+      languageMenu.scrolling=false;
+      languageMenu.y0=e.touches[0].pageY;//osk.lgList.childNodes[0].scrollTop;
+
+      lockBodyScroll();
+    };
+
+    //TODO: Still drags Android background sometimes (not consistently)
+    // Touchmove drags the list and prevents release from selecting the language
+    const touchMove=function(e: TouchEvent) {
+      e.stopImmediatePropagation();
+      var scroller=<HTMLElement>languageMenu.lgList.childNodes[0],
+          yMax=scroller.scrollHeight-scroller.offsetHeight,
+      y, dy;
+
+      // TS does not have a standard definition for TouchEvent.pageY.
+      //@ts-ignore
+      if("undefined" != typeof e.pageY) {
+        //@ts-ignore
+        y = e.pageY;
+      } else if("undefined" != typeof e.touches) {
+        y = e.touches[0].pageY;
+      } else {
+        return;
+      }
+
+      dy=y-languageMenu.y0;
+
+      // Scroll up (show later listed languages)
+      if(dy < 0) {
+        if(scroller.scrollTop >= yMax-1) {
+          languageMenu.y0=y;
+        }
+        // Scroll down (show earlier listed languages)
+      } else if(dy > 0) {
+        if(scroller.scrollTop < 2) {
+          languageMenu.y0=y;
+        }
+        // Dont' scroll - can happen if changing scroll direction
+      } else {
+        return;
+      }
+
+      // Disable selected language if drag more than 5px
+      if(dy < -5 || dy > 5) {
+        languageMenu.scrolling=true;
+        this.className=this.className.replace(/\s*selected/,'');
+        languageMenu.y0=y;
+      }
+      return true;
+    };
```

Comment:
The reason I'm making such a fuss regarding lack of arrow functions and `this` - see line 518.